### ISSUE TITLE
Possibly wrong offsets

### ISSUE THREAD

**Issue by AlexeyGorbushin** [akusio/KernBypass-Public#4]
This tool is unable to bypass JB detection even checking of the '/' is performed.
Original snapshot is mounted 
`orig-fs@/dev/disk0s1s1 on /private/var/MobileSoftwareUpdate/mnt1 (apfs, local, nosuid, read-only, journaled)`
but checking the files in /usr/lib doesn't work. 

The changerootfs tool is able to get notifications
```
changerootfs[1023:9365] receive notify {
    Pid = 1061;
}
vp = 0xffffffe003925078, usecount = 61444, iocount = 61442
```
but further kernel patching to change root has no effect.
The app still tries to access 
`/dev/disk0s1s1 on / (apfs, local, nosuid, union, journaled, noatime)`

I think the problem is in the offsets which can be different from version to version of iOS
```
static const uint32_t off_p_pid = 0x68;
static const uint32_t off_p_pfd = 0x108;
static const uint32_t off_fd_rdir = 0x40;
static const uint32_t off_fd_cdir = 0x38;
static const uint32_t off_vnode_iocount = 0x64;
static const uint32_t off_vnode_usecount = 0x60;
```
It doesn't work at least on iPhone 8 (D201AP), iOS 13.4.1

**Comment by nicho1asdev** [comment]
It doesn’t seem to work for me either. iPhone XR iOS 13.3.

**Comment by coolymike** [comment]
Doesn't work for me, Apps hang at splash screen image and after a minute or 2 crash.
iPhone X iOS 13.2.2

However, on my iPad Air 2 running iOS 13.3.1 it's fully working as far as I can tell.

**Comment by AlexeyGorbushin** [comment]
@coolymike Yeah, bad offsets can explain it that the user objects in the kernel are corrupted by the wrong offsets which leads to the app crashes.

**Comment by Randomblock1** [comment]
Maybe a tweak incompatibility? I had this same issue but uninstalled Choicy and PicaHide (SnapHide 2) and it worked perfectly.
It could be offsets, but even though it received notifications and mounted mnt1, it wouldn’t work until I uninstalled some tweaks.
I’d recommend using iCleaner to disable everything but zzzzNotifier or whatever this uses and open an app that works with this bypass (like Pokémon Go) to check if it really is more than a conflict.
